### PR TITLE
libproxy 0.5.3

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
need to update to meson build

**Comment by chenrui333** [comment]
- relates to https://github.com/libproxy/libproxy/issues/216

**Comment by miimou** [comment]
* one of these patch fixes the problem in my environment
  * add `meson` option to stop documentation generation
      ```diff
      --- libproxy.rb
      @@ -36,3 +36,3 @@
         def install
      -    system "meson", "setup", "build", *std_meson_args
      +    system "meson", "setup", "build", "-Ddocs=false", *std_meson_args
           system "meson", "compile", "-C", "build", "--verbose"
      ```
  * add patch for `docs/meson.build` to add include path for `gi-docgen`
      ```diff
      --- libproxy.rb
      @@ -28,2 +28,6 @@
         depends_on "libsoup"
      +  
      +  patch :p0, "--- docs/meson.build\n" +
      +  "@@ -34,0 +35 @@\n" +
      +  "+    '--add-include-path=" + Formula["gobject-introspection"].opt_share/"gir-1.0" + "',\n"
       
      ```

* as the log shows, build fails because `gi-docgen` cannot find GIR file for `Gio-2.0`
    ```
      FAILED: docs/libproxy-1.0 
      /opt/homebrew/opt/gi-docgen/bin/gi-docgen generate --quiet --add-include-path=/private/tmp/libproxy-20230521-28479-1b6ohit/libproxy-libproxy-0.5.0/docs/src --config=docs/px.toml --output-dir=docs/libproxy-1.0 --no-namespace-dir --content-dir=/private/tmp/libproxy-20230521-28479-1b6ohit/libproxy-libproxy-0.5.0/docs src/libproxy/Libproxy-1.0.gir
      ERROR: Could not find GIR dependency in the search paths: Gio-2.0
    ```
* the GIR for Gio-2.0 is `{HOMEBREW_PREFIX}/share/gir-1.0/Gio-2.0.gir` which is installed by `gobject-introspection` formula
  * the second patch passes the path to `share/gir-1.0` directory to `gi-docgen`
* but, if we do not need documentation, we do not need this
  * `gi-docgen` is used just for generating documentation of `libproxy`, thus this problem can be avoided by simply skipping build of documentation 
  * this is what the first patch does

File: Formula/libproxy.rb [hunk 1]
```diff
@@ -16,34 +16,27 @@ class Libproxy < Formula
...
-    args = %W[
-      -DPYTHON3_SITEPKG_DIR=#{prefix/Language::Python.site_packages("python3.11")}
-      -DWITH_PERL=OFF
-      -DWITH_PYTHON2=OFF
-    ]
-
-    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
-    system "cmake", "--build", "build"
-    system "cmake", "--install", "build"
+    system "meson", "setup", "build", *std_meson_args
```
Comment (SMillerDev) [inline Formula/libproxy.rb:37]:
```suggestion
    system "meson", "setup", "-Ddocs=false", "build", *std_meson_args
```

**Review by SMillerDev — APPROVED** [review]

**Comment by alebcay** [comment]
Adding back to merge queue due to unrelated audit failures (changes in `brew` to check PyPI package names, `openssl@1.1` reaching deprecation date).

**Pull request merged**